### PR TITLE
kirin: first implementation of newrelic and add some logs

### PR DESCRIPTION
**Pull request by kadhikari**
concerned ticket: https://jira.canaltp.fr/browse/NAVP-698

### CODE REVIEW

File: kirin/api.py [hunk 1]
```diff
@@ -81,3 +82,21 @@ def access_log(response, *args, **kwargs):
...
+    return response
+
+@app.after_request
+def add_info_newrelic(response, *args, **kwargs):
+    try:
+        record_custom_parameter('kirin-request-id', request.id)
+        record_custom_parameter('method', request.method)
+        record_custom_parameter('full_path', request.full_path)
+        record_custom_parameter('url', request.url)
+        record_custom_parameter('status_code', request.status_code)
```
Comment (kinnou02) [inline kirin/api.py]: Most of those things are already tracked by newrelic...

File: kirin/ire/model_maker.py [hunk 2]
```diff
@@ -144,6 +145,7 @@ def build(self, rt_update):
         try:
             root = ElementTree.fromstring(rt_update.raw_data)
         except ElementTree.ParseError as e:
+            record_call('ire', 'failure', reason=str(e))
```
Comment (kinnou02) [inline kirin/ire/model_maker.py]: what?

**Comment by kadhikari** [comment]
test this please

File: kirin/gtfs_rt/model_maker.py [hunk 3]
```diff
@@ -124,19 +127,21 @@ def _get_navitia_vjs(self, trip, data_time):
...
+            record_internal_failure('gtfs-rt', 'missing vj')
 
         return [model.VehicleJourney(nav_vj, since.date()) for nav_vj in navitia_vjs]
 
     def _make_stoptime_update(self, input_st_update, navitia_vj):
         nav_st = self._get_navitia_stop_time(input_st_update, navitia_vj)
 
         if nav_st is None:
-            self.log.debug('impossible to find stop point {} in the vj {}, skipping it'.format(
+            self.log.info('impossible to find stop point {} in the vj {}, skipping it'.format(
```
Comment (pbougue) [inline kirin/gtfs_rt/model_maker.py]: should be a warn also (it's the same level than a missing vj)

File: kirin/gtfs_rt/model_maker.py [hunk 4]
```diff
@@ -124,19 +127,21 @@ def _get_navitia_vjs(self, trip, data_time):
         })
 
         if not navitia_vjs:
```
Comment (pbougue) [inline kirin/gtfs_rt/model_maker.py]: we should also log (and skip I think) if we get multiple vj

File: kirin/gtfs_rt/model_maker.py [hunk 5]
```diff
@@ -127,11 +127,20 @@ def _get_navitia_vjs(self, trip, data_time):
...
-            self.log.warn('impossible to find vj {t} on [{s}, {u}['
+            self.log.info('impossible to find vj {t} on [{s}, {u}['
                           .format(t=vj_source_code,
                                   s=since,
                                   u=until))
             record_internal_failure('gtfs-rt', 'missing vj')
+            return None
+
+        if len(navitia_vjs) > 1:
+            self.log.info('too many  vjs found for {t} on [{s}, {u}['
```
Comment (pbougue) [inline kirin/gtfs_rt/model_maker.py]: minor: 2 spaces "many  vjs"
maybe also log vj ids

File: kirin/api.py [hunk 6]
```diff
@@ -71,8 +72,10 @@ def log_exception(sender, exception):
 
     if isinstance(exception, HTTPException):
         logger.debug(error)
+        record_exception()
     else:
         logger.exception(error)
+        record_exception()
```
Comment (kinnou02) [inline kirin/api.py]: so, we can put it a few line before and no duplicate it

File: kirin/api.py [hunk 7]
```diff
@@ -81,3 +84,15 @@ def access_log(response, *args, **kwargs):
     logger = logging.getLogger('kirin.access')
     logger.info('"%s %s" %s', request.method, request.full_path, response.status_code)
     return response
+
+
+@app.after_request
+def add_info_newrelic(response, *args, **kwargs):
```
Comment (kinnou02) [inline kirin/api.py]: still useless

**Pull request merged**